rosidl_typesupport_connext_c/package.xml

```diff
@@ -10,14 +10,14 @@
   <buildtool_depend>ament_cmake</buildtool_depend>
   <buildtool_depend>connext_cmake_module</buildtool_depend>
   <buildtool_depend>rosidl_cmake</buildtool_depend>
-  <buildtool_depend>rosidl_generator_c</buildtool_depend>
+  <buildtool_depend>rosidl_runtime_c</buildtool_depend>
```

Comment:
The package doesn't seem to b used in the `CMakeLists.txt`. Why is it needed at build time?

Reply:
https://github.com/ros2/rosidl_typesupport_connext/pull/49/commits/10197e939bac8025991db110642145638f8d9182

Reply:
Can you please add a sentence of rational to the commit hash. Is your point that the dependency was unnecessary before and removing it is unrelated to this PR?

Reply:
That doesn't seem right imo:

https://github.com/ros2/rosidl_typesupport_connext/blob/1284cbce6338e52fd884cf89cdca69309a3fdb61/rosidl_typesupport_connext_c/CMakeLists.txt#L57-L59

https://github.com/ros2/rosidl_typesupport_connext/blob/1284cbce6338e52fd884cf89cdca69309a3fdb61/rosidl_typesupport_connext_c/src/wstring_conversion.cpp#L15

https://github.com/ros2/rosidl_typesupport_connext/blob/1284cbce6338e52fd884cf89cdca69309a3fdb61/rosidl_typesupport_connext_c/include/rosidl_typesupport_connext_c/wstring_conversion.hpp#L20

Reply:
This dependency is only being exported

https://github.com/ros2/rosidl_typesupport_connext/blob/1284cbce6338e52fd884cf89cdca69309a3fdb61/rosidl_typesupport_connext_c/CMakeLists.txt#L50

It's not being used in any other place. Probably this is wrong, right? It does make sense to add here rosidl_runtime_c

https://github.com/ros2/rosidl_typesupport_connext/blob/1284cbce6338e52fd884cf89cdca69309a3fdb61/rosidl_typesupport_connext_c/CMakeLists.txt#L65-L68

As you indicated above a file with this headers is added to the library

Reply:
> As you indicated above a file with this headers is added to the library

Where is the necessary build dependency for that then? The latest commit (10197e939bac8025991db110642145638f8d9182) doesn't seem to address my point.

Reply:
I removed these lines because my undestanding is that rosidl_runtime_c is not a `<buildtool_depend>` but a `<build_depend>`. Is it that correct?

Reply:
> I removed these lines because my undestanding is that rosidl_runtime_c is not a `<buildtool_depend>` but a `<build_depend>`. Is it that correct?

Yes, what you are stating in the comment is correct. But where is the `build_depend` in the manifest?

Reply:
So, then I need to include the rosidl_runtime_c as a `build_depend` in the package.xml. And In the CMakeLIsts.txt add these two missing things:

```
find_package(rosidl_runtime_c REQUIRED)
```

and add the dependecy here

https://github.com/ros2/rosidl_typesupport_connext/blob/1284cbce6338e52fd884cf89cdca69309a3fdb61/rosidl_typesupport_connext_c/CMakeLists.txt#L65-L68

right?

Reply:
Yes, I would think so.

Reply:
Included here https://github.com/ros2/rosidl_typesupport_connext/pull/49/commits/b0fd3672660c8e207220e27732a2f6ab99e83123